graphql-dgs-spring-boot-micrometer/src/main/kotlin/com/netflix/graphql/dgs/metrics/micrometer/dataloader/DgsDataLoaderInstrumentationProvider.kt

```diff
@@ -2,94 +2,61 @@ package com.netflix.graphql.dgs.metrics.micrometer.dataloader
-    private val mappedBatchLoaderWithContextClasses = mutableMapOf<String, Constructor<out MappedBatchLoaderWithContext<*, *>>>()
-
-    override fun provide(original: BatchLoader<*, *>, name: String): BatchLoader<*, *> {
-        return batchLoaderClasses.getOrPut(name) {
-            val withBinders =
-                MethodDelegation
-                    .withDefaultConfiguration()
-                    .withBinders(Pipe.Binder.install(Forwarder::class.java))
-                    .to(BatchLoaderInterceptor(original, name, meterRegistrySupplier.get()))
-            ByteBuddy()
-                .subclass(BatchLoader::class.java)
-                .method(ElementMatchers.named("load"))
-                .intercept(withBinders)
-                .make()
-                .load(javaClass.classLoader)
-                .loaded
-                .getDeclaredConstructor()
-        }.newInstance()
-    }
-
-    override fun provide(original: BatchLoaderWithContext<*, *>, name: String): BatchLoaderWithContext<*, *> {
-        return batchLoaderWithContextClasses.getOrPut(name) {
-            val withBinders = MethodDelegation.withDefaultConfiguration()
-                .withBinders(Pipe.Binder.install(Forwarder::class.java))
-                .to(BatchLoaderWithContextInterceptor(original, name, meterRegistrySupplier.get()))
-
-            ByteBuddy()
-                .subclass(BatchLoaderWithContext::class.java)
-                .method(ElementMatchers.named("load"))
-                .intercept(withBinders)
-                .make()
-                .load(javaClass.classLoader)
-                .loaded
-                .getDeclaredConstructor()
-        }.newInstance()
-    }
-
-    override fun provide(original: MappedBatchLoader<*, *>, name: String): MappedBatchLoader<*, *> {
-        return mappedBatchLoaderClasses.getOrPut(name) {
-            val withBinders = MethodDelegation.withDefaultConfiguration()
-                .withBinders(Pipe.Binder.install(Forwarder::class.java)).to(
-                    MappedBatchLoaderInterceptor(original, name, meterRegistrySupplier.get())
-                )
-
-            ByteBuddy()
-                .subclass(MappedBatchLoader::class.java)
-                .method(ElementMatchers.named("load"))
-                .intercept(withBinders)
-                .make()
-                .load(javaClass.classLoader)
-                .loaded
-                .getDeclaredConstructor()
-        }.newInstance()
-    }
+    private val batchLoaderClasses = mutableMapOf<String, BatchLoader<*, *>>()
+    private val batchLoaderWithContextClasses = mutableMapOf<String, BatchLoaderWithContext<*, *>>()
+    private val mappedBatchLoaderClasses = mutableMapOf<String, MappedBatchLoader<*, *>>()
+    private val mappedBatchLoaderWithContextClasses = mutableMapOf<String, MappedBatchLoaderWithContext<*, *>>()
+
+    override fun provide(original: BatchLoader<*, *>, name: String): BatchLoader<*, *> =
+        batchLoaderClasses.getOrPut(name) {
```

Comment:
I am not creating a new instance at each access like it was done with ByteBuddy. Not sure if it will cause any issue. Would like to know your opinion on this.

Reply:
We were recently debating why we did that for Bytebuddy, and I couldn't remember the reason (I originally implemented it I think), so it might not have been necessary.
It should be ok like this I think.

Reply:
Great Thanks.